apps/condo/domains/notification/adapters/firebaseAdapter.js

```diff
@@ -118,6 +119,32 @@ class FirebaseAdapter {
         return { title, body }
     }
 
+    /**
+     * Firebase rejects push if any of data fields is not a string, so we should convert all non-string fields to strings
+     * @param data
+     */
+    static prepareData (data = {}) {
+        const result = {}
+        const invalidFields = []
+
+        for (const key in data) {
+            if (data.hasOwnProperty(key)) {
+                result[key] = data[key]
+
+                if (!isString(data[key])) {
+                    result[key] = data[key].toString()
+                    invalidFields.push([key, data[key]])
```

Comment:
Why not Object? Object is more readable after stringifying than an array of arrays.
```suggestion
                    invalidFields[key] = data[key]
```

Reply:
invalidFields is an Array

Reply:
I see, but why not Object instead?

Reply:
actually no profit of having object here, also calculating object emptiness is more expensive then array emptiness :)